sigar_linux_common.go

```diff
@@ -378,15 +378,33 @@ func readFile(file string, handler func(string) bool) error {
 	}
 
 	reader := bufio.NewReader(bytes.NewBuffer(contents))
+	read_line_bytes := new(bytes.Buffer)
+	break_after_line := false
 
 	for {
-		line, _, err := reader.ReadLine()
+		line, prefix, err := reader.ReadLine();
```

Comment:
Should we use [`reader.ReadString('\n')`](https://pkg.go.dev/bufio#Reader.ReadString) instead? We are keeping the whole line in the buffer in any case.
```suggestion
		line, err := reader.ReadString('\n');
```

Or we could maybe use a [bufio.Scanner](https://pkg.go.dev/bufio#Scanner) too.

Reply:
My intention was to not change the used language elements but if you think one of those is better suited, I'll change the patch.

Reply:
I think it would be worth to try with some of the alternatives, as the resulting code might be easier to maintain.

For example with `bufio.Scanner` I think this function could be reduced to something like this (not tested):
```go
func readFile(file string, handler func(string) bool) error {
    reader, err := os.Open(file)
    if err != nil {
        return err
    }
    defer reader.Close()

    scanner := bufio.NewScanner(reader)
    for scanner.Scan() {
        if !handler(scanner.Text()) {
            break
        }
    }
    if err := scanner.Error(); err != nil {
        return err // Current code would return nil in this case, but seems wrong :?
    }
    return nil
}
```

Reply:
This did work on the first try with the new test, i used it in the latest commit. Thanks for your support.